src/main/kotlin/logic/usecase/auth/AuthenticateUserLoginUseCase.kt

```diff
@@ -7,6 +7,15 @@ class AuthenticateUserLoginUseCase(
     private val authRepo: AuthenticationRepository
 ) {
     operator fun invoke(username: String, password: String): Result<User> {
+        if (username.isBlank()) {
+            return Result.failure(IllegalArgumentException("Username cannot be empty"))
+        }
+        if (password.isBlank()) {
+            return Result.failure(IllegalArgumentException("Password cannot be empty"))
+        }
+        if (!username.matches(Regex("^[A-Za-z0-9+_.-]+@(.+)$"))) {
+            return Result.failure(IllegalArgumentException("Invalid email format"))
+        }
```

Comment:
This is only a login function, we don't care what the input is as long as it's valid (decided by the UI), we simply query the given input against our database.

This kind of validation can be benefited from in a register-like use-case.